File: android/app3mvc/app/src/main/java/edu/curso/app3_mvc/ContatoActivity.java
package edu.curso.app3_mvc;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class ContatoActivity extends AppCompatActivity {

    private ContatoControl control;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.contato_activity);
        control = new ContatoControl( this );
        control.bind();
    }

}

File: android/agenda-contatos/app/src/main/java/edu/curso/agenda/ui/ContatoActivity.kt
package edu.curso.agenda.ui

import android.os.Bundle
import android.widget.ArrayAdapter
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import edu.curso.agenda.ContatoActivityViewModel
import edu.curso.agenda.databinding.ContatoActivityBinding
import edu.curso.agenda.entity.Contato

class ContatoActivity: AppCompatActivity() {

    private lateinit var binding : ContatoActivityBinding
    private lateinit var viewModel : ContatoActivityViewModel
    override fun onCreate(bundle : Bundle?) {
        super.onCreate(bundle)
        viewModel = ViewModelProvider(this)[ContatoActivityViewModel::class.java]
        binding =  ContatoActivityBinding.inflate( layoutInflater  )
        setContentView(binding.root)
        binding.contato = viewModel.getContato()
        binding.apply {

            btnAdicionar.setOnClickListener {
                viewModel.addContato()
            }
        }
    }
}